### PR TITLE
Feature/view performance

### LINKED ISSUE #6: Improve selectView performance
- [x] select view with 100 parents and 1000 children runs below 10ms on local machines
- [x] select view with 100 parents and 1000 children runs below 25ms in workflows

### PR DESCRIPTION
**Pull request by Berndy**
resolves #6 

## Changes
- Added cheaper util function versions of ramda functions
    - mutAssocPath: mutates the given object and returns it
    - getPath: gets the instance at the path
    - getPathOr: gets the instance at the path or the given fallback 
- Reworked selectPathFromStack to not use expensive immutable ramda functions as much where it is not necessary
    - this accelerates every select function of got-store
- Reworked useResult to return an over function as third element to make executing setter exclusively with functions faster
- Reworked selectView to not use expensive immutable ramda functions as much where it is not necessary
    - now using over function to mutate result instead of setter
    - now constructing bag as one object instead of associng every single bag prop, which is expensive
    - now using faster function to check if props should be included

### CODE REVIEW

**Review by AlphaBroke — APPROVED** [review]
Nice 1

File: packages/got-util/src/index.js [hunk 1]
```diff
@@ -137,3 +145,36 @@ export const toPromise = observable => new Promise((resolve, reject) => {
...
+        } else {
+            return or;
+        }
+    }
+    return obj;
+};
+
+export const getPath = (path, input) => getPathOr(undefined, path)(input);
+
+export const mutAssocPath = (path, val) => input => {
```
Comment (vinnichase) [inline packages/got-util/src/index.js]: assocPathMut or assocPathMutate

File: packages/got-store/src/tests/views.spec.js [hunk 2]
```diff
@@ -759,6 +760,75 @@ describe('store:Views', () => {
...
+                const {
+                    store: {
+                        selectView,
+                        add,
+                    },
+                    select,
+                } = createTestStore({ });
+
+                // generate debug data
+                for (let i = 0; i < 100; i += 1) {
```
Comment (vinnichase) [inline packages/got-store/src/tests/views.spec.js]: Exclude to util as generate random graph

**Review by vinnichase — CHANGES_REQUESTED** [review]

**Review by vinnichase — APPROVED** [review]

File: packages/got-util/src/index.d.ts [hunk 3]
```diff
@@ -105,3 +114,20 @@ export declare const forEachConfObj: (
 export declare const toPromise: (
     observable: { subscribe: (subscriber: Subscriber<any>) => void }
     ) => Promise<any[]>;
+
+/**
```
Comment (andvrs) [inline packages/got-util/src/index.d.ts:118]: why 2

**Review by andvrs — APPROVED** [review]
Very nice job. Keep it up

**Pull request merged**